Add to support patches provided by ONIE for linux-custom

Copyright (C) 2014 david_yang <david_yang@accton.com>

SPDX-License-Identifier:     GPL-2.0

diff --git a/scripts/build/kernel/linux.sh b/scripts/build/kernel/linux.sh
index 5f32680..50a28bf 100644
--- a/scripts/build/kernel/linux.sh
+++ b/scripts/build/kernel/linux.sh
@@ -75,11 +75,9 @@ do_kernel_extract() {
     # tarball; in either case, we need to extract
     CT_Extract "linux-${CT_KERNEL_VERSION}"
 
-    # If using a custom tarball, no need to patch
-    if [ "${CT_KERNEL_LINUX_CUSTOM}" = "y" ]; then
-        return 0
-    fi
-    CT_Patch "linux" "${CT_KERNEL_VERSION}"
+    # To support patches provided by ONIE for linux-custom
+    ln -s ${CT_SRC_DIR}/linux-custom ${CT_SRC_DIR}/linux-${LINUX_RELEASE}
+    CT_Patch "linux" "${LINUX_RELEASE}"
 }
 
 # Wrapper to the actual headers install method
